target/linux/ramips/patches-5.10/415-add-support-for-eon-en25qh256a.patch

```diff
@@ -0,0 +1,63 @@
+From 6abef37d16d0c570ef5a149e63762fba2a30804b Mon Sep 17 00:00:00 2001
+From: "Leon M. George" <leon@georgemail.eu>
+Date: Wed, 30 Mar 2022 16:16:56 +0200
+Subject: [PATCH] mtd: spi-nor: support eon en25qh256a variant
+
+The en25qh256a requires 4-byte_addr_mode to operate.
+Otherwise, it appears to be identical to the en25qh256.
+
+Signed-off-by: Leon M. George <leon@georgemail.eu>
+---
+ drivers/mtd/spi-nor/eon.c | 11 +++++++++++
+ 1 file changed, 11 insertions(+)
+
+--- a/drivers/mtd/spi-nor/eon.c
++++ b/drivers/mtd/spi-nor/eon.c
+@@ -8,6 +8,37 @@
+ 
+ #include "core.h"
+ 
++#define DEBUG 1
++
++static int
++en25qh256_post_bfpt_fixups(struct spi_nor *nor,
++                           const struct sfdp_parameter_header *bfpt_header,
++                           const struct sfdp_bfpt *bfpt,
++                           struct spi_nor_flash_parameter *params)
++{
++pr_err("isn't called :-(\n");
++pr_err("POST ADDRESS_BYTES %lu\n", bfpt->dwords[BFPT_DWORD(1)] & BFPT_DWORD1_ADDRESS_BYTES_MASK);
++pr_err("POST NOR_FLAGS %u\n", nor->flags);
++pr_err("POST ADDR_WIDTH %u\n", (unsigned int) nor->addr_width);
++
++	// if (bfpt->dwords[BFPT_DWORD(1)] & BFPT_DWORD1_ADDRESS_BYTES_MASK)
++	// 	is there anything else to do?
++
++	return 0;
++}
++
++static void en25qh256_default_init(struct spi_nor *nor)
++{
++pr_err("debug is working %u\n", nor->flags);
++	// TODO maybe set from post_bfpt_fixup
++	nor->params->set_4byte_addr_mode = spi_nor_set_4byte_addr_mode;
++}
++
++static const struct spi_nor_fixups en25qh256_fixups = {
++	.default_init = en25qh256_default_init,
++	.post_bfpt = en25qh256_post_bfpt_fixups,
++};
++
+ static const struct flash_info eon_parts[] = {
+ 	/* EON -- en25xxx */
+ 	{ "en25f32",    INFO(0x1c3116, 0, 64 * 1024,   64, SECT_4K) },
+@@ -24,7 +55,8 @@ static const struct flash_info eon_parts
+ 	{ "en25qh64",   INFO(0x1c7017, 0, 64 * 1024,  128,
+ 			     SECT_4K | SPI_NOR_DUAL_READ) },
+ 	{ "en25qh128",  INFO(0x1c7018, 0, 64 * 1024,  256, 0) },
+-	{ "en25qh256",  INFO(0x1c7019, 0, 64 * 1024,  512, 0) },
++	{ "en25qh256",  INFO(0x1c7019, 0, 64 * 1024,  512, 0)
++			     .fixups = &en25qh256_fixups, },
```

Comment:
no comma after fixups, only after the brace

I do not know whether that comma causes a problem or not, for all we know it could be setting the next value of the struct to be null